compiler/rustc_mir_transform/src/const_prop_lint.rs

```diff
@@ -63,6 +63,10 @@ impl<'tcx> MirLint<'tcx> for ConstProp {
             return;
         }
 
+        if body.tainted_by_errors.is_some() {
```

Comment:
This should probably be checked at the top of the function, no need to do the check so late.

Now I wonder if this means we can get rid of some of the other ConstPropNonsense throws...